src/main/java/uk/gov/hmcts/reform/et/syaapi/service/StoredRespondToApplicationSubmitService.java

```diff
@@ -28,11 +40,131 @@ public class StoredRespondToApplicationSubmitService {
+        }
+
+        GenericTseApplicationType appType = appToModify.getValue();
+        appType.setClaimantResponseRequired(NO);
+        appType.setApplicationState(STORED);
+
+        // Add response to RespondStoredCollection
+        setRespondentApplicationWithResponse(request, appType, caseData, caseDocumentService);
+
+        // Update pdf
+        createAndAddPdfOfResponse(authorization, request, caseData, appType);
+
+        // Send email
+        NotificationService.CoreEmailDetails details = notificationService.formatCoreEmailDetails(caseData, caseId);
+        notificationService.sendStoredEmailToClaimant(details, appType.getType());
+
+        return caseService.submitUpdate(
+            authorization, caseId, caseDetailsConverter.caseDataContent(startEventResponse, caseData), caseTypeId);
+    }
+
+    private void setRespondentApplicationWithResponse(RespondToApplicationRequest request,
+                                                            GenericTseApplicationType appType,
+                                                            CaseData caseData,
+                                                            CaseDocumentService caseDocumentService) {
+        if (CollectionUtils.isEmpty(appType.getRespondStoredCollection())) {
+            appType.setRespondStoredCollection(new ArrayList<>());
+        }
+
+        TseRespondType responseToAdd = request.getResponse();
+        responseToAdd.setDate(TseApplicationHelper.formatCurrentDate(LocalDate.now()));
+        responseToAdd.setFrom(CLAIMANT_TITLE);
+        responseToAdd.setDateTime(getCurrentDateTime());
+        responseToAdd.setApplicationType(appType.getType());
+
+        // Supporting Material File
+        if (request.getSupportingMaterialFile() != null) {
+            DocumentTypeItem documentTypeItem = caseDocumentService.createDocumentTypeItem(
+                CLAIMANT_CORRESPONDENCE_DOCUMENT,
+                request.getSupportingMaterialFile()
+            );
+            documentTypeItem.getValue().setShortDescription("Response to " + appType.getType());
+
+            responseToAdd.setSupportingMaterial(new ArrayList<>());
+            responseToAdd.getSupportingMaterial().add(documentTypeItem);
+
+            String applicationDoc = getApplicationDoc(appType);
+            String extension = FilenameUtils.getExtension(request.getSupportingMaterialFile().getDocumentFilename());
+            String docName = "Application %s - %s - Attachment.%s".formatted(
+                appType.getNumber(),
+                appType.getType(),
+                extension
+            );
+            request.getSupportingMaterialFile().setDocumentFilename(docName);
+            documentTypeItem = caseDocumentService.createDocumentTypeItem(applicationDoc,
+                                                                          request.getSupportingMaterialFile());
+
+            if (caseData.getDocumentCollection() == null) {
```

Comment:
Whilst this works, for consistency if you consider here using the null safe CollectionUtils.isEmpty() like on line 139?